### PR TITLE
sysinfo improvements

### PR DESCRIPTION
**Pull request by apahim**
- Print debug information on missing `files`.
- Include a journalctl diff in tests sysinfo results.
- Make tests to respect sysinfo enable/disabled configuration.

### CODE REVIEW

File: avocado/core/sysinfo.py [hunk 1]
```diff
@@ -102,6 +102,8 @@ def run(self, logdir):
                 shutil.copyfile(self.path, os.path.join(logdir, self.logf))
             except IOError:
                 log.debug("Not logging %s (lack of permissions)", self.path)
+        else:
+            log.debug("Not logging %s (file does not exist)", self.path)
```
Comment (clebergnu) [inline avocado/core/sysinfo.py]: OK, as simple (and effective) as it gets.

File: avocado/core/sysinfo.py [hunk 2]
```diff
@@ -213,6 +214,45 @@ def stop(self):
         return retcode
 
 
+class JournalctlWatcher(Collectible):
+
+    """
+    Track the content of systemd journal into a compressed file.
+
+    :param logf: Basename of the file where output is logged (optional).
```
Comment (clebergnu) [inline avocado/core/sysinfo.py]: The `:param logf:` part of the docstring should be on `__init__()`.

File: avocado/core/sysinfo.py [hunk 3]
```diff
@@ -213,6 +214,45 @@ def stop(self):
...
+    """
+
+    def __init__(self, logf=None):
+        if not logf:
+            logf = 'journalctl.gz'
+
```
Comment (clebergnu) [inline avocado/core/sysinfo.py]: PEP8 did not complain about this empty line, but I am complaining... :smile: It's really minor, but IMHO feels a bit inconsistent.

File: avocado/core/sysinfo.py (hunk 2 above)
Comment (clebergnu) [inline avocado/core/sysinfo.py]: Oh, now I notice that `LogFile` does the same (documents `__init__` parameters on the class docstring).  Anyway, I really think it's not the best place for it.

File: avocado/core/sysinfo.py (hunk 3 above)
Comment (clebergnu) [inline avocado/core/sysinfo.py]: And again previous art on `sysinfo` makes my comment at least weird.  `LogWatcher` also adds this empty line... And others don't.  Small little style issues that my raise big red signs on my eyes... sigh.

Comment (clebergnu) [inline avocado/core/sysinfo.py]: Anyway, don't *focus* on those style comments.  Address them only if you feel like it.

File: avocado/core/sysinfo.py [hunk 4]
```diff
@@ -213,6 +214,45 @@ def stop(self):
...
+        super(JournalctlWatcher, self).__init__(logf)
+        self.cursor = self._get_cursor()
+
+    def _get_cursor(self):
+        try:
+            cmd = 'journalctl --lines 1 --output json'
+            result = process.system_output(cmd, verbose=False)
+            last_record = json.loads(result)
+            return last_record['__CURSOR']
```
Comment (clebergnu) [inline avocado/core/sysinfo.py]: OK, makes sense.

File: avocado/core/sysinfo.py [hunk 5]
```diff
@@ -213,6 +214,45 @@ def stop(self):
...
+        except Exception as e:
+            log.debug("Journalctl collection failed: %s", e)
+
+    def run(self, logdir):
+        if self.cursor:
```
Comment (clebergnu) [inline avocado/core/sysinfo.py]: Note to self: gets implicitly set to `None` if exception is raised and `_get_cursor()` returns `None`.

File: avocado/core/sysinfo.py [hunk 6]
```diff
@@ -213,6 +214,45 @@ def stop(self):
...
+            try:
+                cmd = 'journalctl --quiet --after-cursor %s' % self.cursor
```
Comment (clebergnu) [inline avocado/core/sysinfo.py]: OK, works when tested manually.

File: avocado/core/sysinfo.py [hunk 7]
```diff
@@ -442,6 +482,8 @@ def _set_collectibles(self):
         except ValueError as details:
             log.info(details)
 
+        self.end_test_collectibles.add(JournalctlWatcher())
```
Comment (clebergnu) [inline avocado/core/sysinfo.py]: OK.

File: selftests/unit/test_sysinfo.py [hunk 8]
```diff
@@ -85,8 +85,8 @@ def test_logger_test_hooks(self):
         job_postdir = os.path.join(testdir, 'post')
         self.assertTrue(os.path.isdir(job_postdir))
         # By default, there are no post test files
-        self.assertLess(len(os.listdir(job_postdir)), 2,
-                        "Post dir can contain 0-1 files depending on whether "
+        self.assertLess(len(os.listdir(job_postdir)), 3,
```
Comment (clebergnu) [inline selftests/unit/test_sysinfo.py]: OK, makes sense.

File: selftests/functional/test_export_variables.py [hunk 9]
```diff
@@ -51,7 +51,7 @@ def setUp(self):
 
     def test_environment_vars(self):
         os.chdir(basedir)
-        cmd_line = './scripts/avocado run --job-results-dir %s --sysinfo=off %s' % (self.tmpdir, self.script.path)
+        cmd_line = './scripts/avocado run --job-results-dir %s --sysinfo=on %s' % (self.tmpdir, self.script.path)
```
Comment (clebergnu) [inline selftests/functional/test_export_variables.py]: Yep, nice one!

**Comment by clebergnu** [comment]
Yep, a `/home/cleber/avocado/job-results/job-2016-09-09T09.35-0108014/test-results/1-passtest.py:PassTest.test/sysinfo/post/journalctl.gz` was properly generated.  Let'me try with some test that will actually generate system log activity.

**Comment by apahim** [comment]
@clebergnu `slepptenmin.py` and restart some systemctl service ;)

**Comment by clebergnu** [comment]
Yep, that's what I'm at.  BTW, let me know if you think this makes sense:

https://github.com/clebergnu/avocado/commit/cef85b252a266fd440085d53032a2a141b517d7f

Just stumbled upon while doing this review.

**Comment by apahim** [comment]
Yes, great catch indeed.

File: avocado/core/sysinfo.py (hunk 3 above)
Comment (ldoktor) [inline avocado/core/sysinfo.py]: Well a style-fix commit would be useful for this file... (before-after)

**Comment by clebergnu** [comment]
OK, so my functional test passes, and I'll consider that you have ACK'd that one extra patch of mine (right?) and merge them.

OK?

**Comment by apahim** [comment]
Sure.

File: avocado/core/sysinfo.py (hunk 3 above)
Comment (clebergnu) [inline avocado/core/sysinfo.py]: Yep, but since this style fix would touch other code also, we can do it after this.

**Comment by clebergnu** [comment]
Merged with the additional commit. Thanks!

**Pull request closed without merging**